File: net/pdfjet/CircleAnnotation.cs
/**
 * CircleAnnotation.cs
 *
 * Copyright (c) 2025 PDFjet Software
 * Licensed under the MIT License. See LICENSE file in the project root.
 */
using System;

namespace PDFjet.NET {
public class CircleAnnotation : BaseAnnotation {
    public CircleAnnotation() {
        base.annotationType = Annotation.Circle;
    }

    public void Rotate(double degrees) {
        float[] center = container.GetRotationCenter();
        float[] xy1 = Container.RotateAroundCenter(x1, y1, center, degrees);
        float[] xy2 = Container.RotateAroundCenter(x2, y2, center, degrees);
        this.x1 = xy1[0];
        this.y1 = xy1[1];
        this.x2 = xy2[0];
        this.y2 = xy2[1];
    }
}
}

File: com/pdfjet/CircleAnnotation.java
/**
 * CircleAnnotation.java
 *
 * Copyright (c) 2025 PDFjet Software
 * Licensed under the MIT License. See LICENSE file in the project root.
 */
package com.pdfjet;

public class CircleAnnotation extends BaseAnnotation {
    public CircleAnnotation() {
        super.annotationType = Annotation.Circle;
    }

    public void rotate(double degrees) {
        float[] center = container.getRotationCenter();
        super.point1 = Container.rotateAroundCenter(super.point1, center, degrees);
        super.point2 = Container.rotateAroundCenter(super.point2, center, degrees);
    }
}
